docs/src/explanations/moduleprefix.md

```diff
@@ -0,0 +1,117 @@
+# Module Prefixing
+
+Chisel supports a feature called module prefixing.
+Module prefixing allows you to create namespaces in the Verilog output of your design.
+They are especially useful for when you want to name a particular subsystem of your design,
+and you want to make it easy to identify which subsystem a file belongs to by its name.
+
+We can open a module prefix block using `withModulePrefix`:
+
+```scala mdoc:silent
+import chisel3._
+
+class Top extends Module {
+  withModulePrefix("Foo") {
+    // ...
+  }
+}
+```
+
+All modules defined inside of this block, whether an immediate submodule or a descendent, will be given a prefix `Foo`.
+(The prefix is separated by an underscore `_`).
+
+For example, suppose we write the following:
+
+```scala mdoc:silent
+import chisel3._
+
+class Top extends Module {
```

Comment:
```suggestion
```scala mdoc:silent:reset
import chisel3._

class Top extends Module {
```
Mdoc is failing because you define top multiple times, I think you just need to reset each block.